src/qibolab/platform.py

```diff
@@ -214,22 +169,49 @@ def disconnect(self):
                 instrument.disconnect()
         self.is_connected = False
 
-    def _execute(self, sequence, options, **kwargs):
+    def _apply_config_updates(
+        self, updates: list[dict[str, Config]]
+    ) -> dict[str, Config]:
+        """Apply given list of config updates to the default configuration and
+        return the updated config dict.
+
+        Args:
+            updates: list of updates, where each entry is a dict mapping component name to new config. Later entries
+                     in the list override earlier entries (if they happen to update the same thing).
+        """
+        components = self.components.copy()
+        for update in updates:
+            for name, cfg in update.items():
+                if name not in components:
+                    raise ValueError(
+                        f"Cannot update configuration for unknown component {name}"
+                    )
+                if type(cfg) is not type(components[name]):
+                    raise ValueError(
+                        f"Configuration of component {name} with type {type(components[name])} cannot be updated with configuration of type {type(cfg)}"
+                    )
+                components[name] = cfg
+        return components
```

Comment:
The reason it is implemented like this is that config classes are frozen. So a user attempting to update a certain property is supposed to get the config from platform, and do replace(config, update). The user is not supposed to manually write config properties as strings in the update dictionary.

Reply:
No, but as you see in the suggestion above, I'm proposing to do the `replace(config, update)` ourselves, since this is an operation repeated over and over.

If the user does not manually write the properties as strings in the `updates` dictionary (the parameter of `_apply_config_updates`), he will manually write the properties as strings in the `update` dictionary passed to `replace(config, update)`.
*And* then manually apply `replace`.
I'm just proposing to embed the second step, to avoid the user to apply over and over (also because the user might be Qibocal, and it would be handy to generate update dictionaries, with just the updated parameter)